frontend/src/components/DashboardView/ChartBlock.tsx

```diff
@@ -0,0 +1,417 @@
+import { useEffect, useState } from 'react';
+import {
+  Box,
+  makeStyles,
+  Typography,
+  CircularProgress,
+  Button,
+} from '@material-ui/core';
+import { DashboardChartConfig, AdminLevelType } from 'config/types';
+import { useChartForm, useChartData } from 'utils/chart-hooks';
+import Chart from 'components/Common/Chart';
+import {
+  ChartLayerSelector,
+  ChartDateRangeSelector,
+  ChartLocationSelector,
+} from 'components/Common/ChartFormComponents';
+import { useSafeTranslation } from 'i18n';
+
+type ChartBlockMode = 'edit' | 'preview';
```

Comment:
At some point it may make sense to consolidate these types to a single version that’s shared among blocks, but definitely not a blocker.